File: tests/case/case.cpp
#include <iostream>
#include <string>

int main() {
    int day;
    int score;
    
    day = 3;
    switch (day) {
        case 1:
            std::cout << "Monday" << std::endl;
            break;
        case 2:
            std::cout << "Tuesday" << std::endl;
            break;
        case 3:
            std::cout << "Wednesday" << std::endl;
            break;
        case 4:
            std::cout << "Thursday" << std::endl;
            break;
        case 5:
            std::cout << "Friday" << std::endl;
            break;
        case 6:
            std::cout << "Saturday" << std::endl;
            break;
        case 7:
            std::cout << "Sunday" << std::endl;
            break;
        default:
            std::cout << "Invalid day" << std::endl;
            break;
    }
    score = 85;
    switch (score) {
        case 90:
        case 91:
        case 92:
        case 93:
        case 94:
        case 95:
        case 96:
        case 97:
        case 98:
        case 99:
            std::cout << 'A' << std::endl;
            break;
        case 80:
        case 81:
        case 82:
        case 83:
        case 84:
        case 85:
        case 86:
        case 87:
        case 88:
        case 89:
            std::cout << 'B' << std::endl;
            break;
        case 70:
        case 71:
        case 72:
        case 73:
        case 74:
        case 75:
        case 76:
        case 77:
        case 78:
        case 79:
            std::cout << 'C' << std::endl;
            break;
        default:
            std::cout << 'F' << std::endl;
            break;
    }
    day = 6;
    switch (day) {
        case 1:
            std::cout << "Weekday" << std::endl;
            break;
        case 2:
            std::cout << "Weekday" << std::endl;
            break;
        case 3:
            std::cout << "Weekday" << std::endl;
            break;
        case 4:
            std::cout << "Weekday" << std::endl;
            break;
        case 5:
            std::cout << "Weekday" << std::endl;
            break;
    }
    day = 1;
    while ((day <= 3)) {
        switch (day) {
            case 1:
                std::cout << "first" << std::endl;
                break;
            case 2:
                std::cout << "second" << std::endl;
                break;
            case 3:
                std::cout << "third" << std::endl;
                break;
        }
        day = (day + 1);
    }
    return 0;
}

File: tests/case/case.pas
program CaseTest;

var
  day: integer;
  score: integer;

begin
  { простой case с числами }
  day := 3;
  case day of
    1: writeln('Monday');
    2: writeln('Tuesday');
    3: writeln('Wednesday');
    4: writeln('Thursday');
    5: writeln('Friday');
    6: writeln('Saturday');
    7: writeln('Sunday');
    else writeln('Invalid day')
  end;

  { case с несколькими значениями в одной ветке }
  score := 85;
  case score of
    90, 91, 92, 93, 94, 95, 96, 97, 98, 99: writeln('A');
    80, 81, 82, 83, 84, 85, 86, 87, 88, 89: writeln('B');
    70, 71, 72, 73, 74, 75, 76, 77, 78, 79: writeln('C');
    else writeln('F')
  end;

  { case без ветки else }
  day := 6;
  case day of
    1: writeln('Weekday');
    2: writeln('Weekday');
    3: writeln('Weekday');
    4: writeln('Weekday');
    5: writeln('Weekday')
  end;

  { case в цикле }
  day := 1;
  while day <= 3 do
  begin
    case day of
      1: writeln('first');
      2: writeln('second');
      3: writeln('third')
    end;
    day := day + 1
  end
end.
